packages/expo-image/android/src/main/java/expo/modules/image/ExpoImageViewWrapper.kt

```diff
@@ -233,74 +236,82 @@ class ExpoImageViewWrapper(context: Context, appContext: AppContext) : ExpoView(
     target: ImageViewWrapperTarget,
     resource: Drawable,
     isPlaceholder: Boolean = false
-  ) {
-    val transitionDuration = (transition?.duration ?: 0).toLong()
+  ) =
+    // The "onResourceReady" function will be triggered when the new resource is available by the Glide.
+    // According to the Glide documentation (https://bumptech.github.io/glide/doc/debugging.html#you-cant-start-or-clear-loads-in-requestlistener-or-target-callbacks),
+    // it's not advisable to clear the Glide target within the stack frame.
+    // To avoid this, a new runnable is posted to the front of the main queue, which can then clean or create targets.
+    // This ensures that the "onResourceReady" frame of the Glide code will be discarded, and the internal state can be altered once again.
+    // Normally, using "postAtFrontOfQueue" can lead to issues such as message queue starvation, ordering problems, and other unexpected consequences.
+    // However, in this case, it is safe to use as long as nothing else is added to the queue.
+    // The intention is simply to wait for the Glide code to finish before the content of the underlying views is changed during the same rendering tick.
+    mainHandler.postAtFrontOfQueue {
```

Comment:
Only this was changed. The rest is here because of the different formatting which has to be applied.